### PR TITLE
Task4 - REST support

### PR DESCRIPTION
**Pull request by Irenkia**
Service should support all operations described in the lesson:
1)    Add new invoice (POST)
2)    List all existing invoices (GET)
3)    Return details of single invoice (GET)
4)    Update existing invoice (PUT)
5)    Delete existing invoice (DELETE)

### CODE REVIEW

File: app/build.gradle
Comment (PatrykZimnowoda) [inline app/build.gradle:1]: Czesc na potrzeby zadania  Marek podeslal mi ten scrypt zeby nie zmieniac nic w Jacoco
 afterEvaluate { // TODO [PK] remove after adding tests in Task #5
        classDirectories.setFrom(files(classDirectories.files.collect {
            fileTree(dir: it, exclude: 'pl/futurecollars/invoicing/controller/**')
        }))
    }

**Review by PatrykZimnowoda — DISMISSED** [review]
Looks good the only thing i would change is in build.gradle i would add 
 afterEvaluate { // TODO [PK] remove after adding tests in Task #5
        classDirectories.setFrom(files(classDirectories.files.collect {
            fileTree(dir: it, exclude: 'pl/futurecollars/invoicing/controller/**')
        }))
    } 
    instead of changing %% coverage :)

File: app/build.gradle
Comment (PatrykZimnowoda) [reply app/build.gradle:1]: afterEvaluate { // TODO [PK] remove after adding tests in Task #5
        classDirectories.setFrom(files(classDirectories.files.collect {
            fileTree(dir: it, exclude: 'pl/futurecollars/invoicing/controller/**')
        }))
    }

Comment (PatrykZimnowoda) [reply app/build.gradle:1]: jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                counter = 'BRANCH'
                minimum = 0.70
            }
        }

        rule {
            limit {
                counter = 'LINE'
                minimum = 0.70
            }
        }
    }
    afterEvaluate { // TODO [PK] remove after adding tests in Task #5
        classDirectories.setFrom(files(classDirectories.files.collect {
            fileTree(dir: it, exclude: 'pl/futurecollars/invoicing/controller/**')
        }))
    }
}

Comment (PatrykZimnowoda) [reply app/build.gradle:1]: here inside JTestCoVerfication :)

**Review by PatrykZimnowoda — APPROVED** [review]
All looking good 
BTW i feeling like u should add changes from task 5 on the New branch instead of alrrady pushing them on this task :)

**Review by marwis1995 — APPROVED** [review]

**Pull request merged**